File: DSA/Ansh-Gulati/Day-1/Day-1_Question-2.cpp
#include <iostream>
using namespace std;

int numberOfValidPairs(int size, int arr[], int k)
{
    int count = 0;

    for (int i = 0; i < size; i++)
    {
        for (int j = i + 1; j < size; j++)
        {
            if ((i < j) && ((i + j) % k == 0))
            {
                count++;
            }
        }
    }

    return count;
}

int main()
{
    int size;
    cout << "Enter the size of the array: ";
    cin >> size;

    int arr[100];

    cout << "Input elements in array: ";

    // For taking input
    for (int i = 0; i < size; i++)
    {
        cin >> arr[i];
    }

    int k;
    cout << "Enter the value of 'k': ";
    cin >> k;

    int result = numberOfValidPairs(size, arr, k);
    cout << "The number of Valid pairs of i,j in the given array are: " << result;
}

File: DSA/Aditi-Sharma/Day1/Day_1_question_2.java
import java.util.*;
public class ValidPairs {

    public static int validPairs(int[] arr, int k) {
        int count = 0;
        for (int j = 1; j < arr.length; j++) {
            for (int i = j - 1; i >= 0; i--) {
                if ((arr[j] + arr[i]) % k == 0) {
                    count++;
                }
            }
        }
        return count;
    }

    public static void main(String[] args) {
        int[] arr = {1, 3, 2, 6, 1, 2};
        int k = 3;
        System.out.println(validPairs(arr, k)); 
    }
}
